### PR TITLE
oqs-test/try_connection.py tests what is in the README.

### PR DESCRIPTION
**Pull request by xvzcf**

### CODE REVIEW

**Review by dstebila — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -112,115 +112,74 @@ On **Ubuntu**, you need to install the following packages:
...
-
-	./configure --with-ssl-dir=<path-to-openssl>/include \
-	            --with-ldflags=-L<path-to-openssl>/lib   \
-	            --with-libs=-lm                          \
-	            --prefix=$OPENSSH_INSTALL                \
-	            --sysconfdir=$OPENSSH_INSTALL            \
-	            --with-liboqs-dir=`pwd`/oqs
-	make 
-	make install
+`OPENSSL_SYS_DIR` does not have to be specified if OpenSSL is present under `usr/`.
```
Comment (baentsch) [inline README.md]: Change to "/usr".

File: README.md [hunk 2]
```diff
@@ -112,115 +112,74 @@ On **Ubuntu**, you need to install the following packages:
...
-
-As not all stock `openssh` tests are passing, be sure to execute `oqs-test/run_tests.sh` instead of simply executing `make tests` to ensure the build was successful.
-
-To execute a connection test with one of the supported quantum-safe algorithms (chosen at random), run `python3 oqs-test/try_connection.py`. If all algorithms should be exercized, pass a parameter to this command, e.g., like this: `python3 oqs-test/try_connection.py all`. Be aware that this test can take a long time due to the number of algorithm combinations available.
+To execute a connection test with a randomly chosen key-exchange and signature algorithm, run `python3 oqs-test/try_connection.py`. If it is desired that all such combinations should be tested, run `python3 oqs-test/try_connection.py all`. Be aware that this test can take a long time due to the number of algorithm combinations available.
```
Comment (baentsch) [inline README.md]: Maybe worthwhile adding a sentence that the "doall" parameter executes each SIG and KEM _exactly once_?

File: README.md [hunk 3]
```diff
@@ -112,115 +112,74 @@ On **Ubuntu**, you need to install the following packages:
...
+To setup quantum-safe authentication, the server (and optionally, the client) need to generate quantum-safe keys. To generate keys for all the OQS algorithms supported by fork, simply run `make tests -e LTESTS=""`.
 
 #### Establishing a quantum-safe SSH connection
 
-In one terminal, run a server:
+In one terminal, run the ssh server:
 
-	<path-to-openssh>/sbin/sshd -p 2222 -d \
+	<path-to-openssh>/sbin/sshd -D \
+	                            -f <absolute-path-to>/regress/sshd_config \
```
Comment (baentsch) [inline README.md]: Is the "\<absolute-path-to\>" wording sufficiently clear? What about stating that this should be the `openssh` build directory? If so, what about replacing with `pwd`, i.e., "...run the ssh server from within the build directory : \<path-to-openssh-install-directory\>/sbin/sshd -D -f \`pwd\`/regress/sshd_config...."

File: README.md [hunk 4]
```diff
@@ -112,115 +112,74 @@ On **Ubuntu**, you need to install the following packages:
...
 	                            -o KexAlgorithms=<KEX> \
-	                            -o AuthorizedKeysFile=<absolute-path-to>/ssh_server/authorized_keys \
 	                            -o HostKeyAlgorithms=ssh-<SIG> \
 	                            -o PubkeyAcceptedKeyTypes=ssh-<SIG> \
-	                            -h <absolute-path-to>/ssh_server/id_<SIG>
+	                            -h <absolute-path-to>/regress/host.<SIG>
```
Comment (baentsch) [inline README.md]: shouldn't this read "...regress/host.ssh-\<SIG\>"?

File: README.md [hunk 5]
```diff
@@ -112,115 +112,74 @@ On **Ubuntu**, you need to install the following packages:
...
 
-	<path-to-openssh>/bin/ssh -p 2222 localhost \
+	<path-to-openssh>/bin/ssh -F <absolute-path-to>/regress/ssh_config \
 	                          -o KexAlgorithms=<KEX> \
 	                          -o HostKeyAlgorithms=ssh-<SIG>\
 	                          -o PubkeyAcceptedKeyTypes=ssh-<SIG> \
-	                          -o StrictHostKeyChecking=no \
 	                          -o PasswordAuthentication=no \
-	                          -i ~/ssh_client/id_<SIG>
+	                          -i regress/<SIG> \
```
Comment (baentsch) [inline README.md]: same as above: SIG ref correct?

**Review by baentsch — CHANGES_REQUESTED** [review]
Please check the specific comments above.
Generally, using the regress data works, of course; but I'd then also suggest adding that 
- host and client keys can be separately generated using `ssh-keygen`
- configuration options (like port etc.) can either be set on the command line or in the respective `ssh_config` and `sshd_config` files in the `regress` folder.

Also, if the whole build documentation is reduced to running the build scripts, I'd suggest adding that "PREFIX" can be set to facilitate a specific `oqs-openssh` install location: I can't imagine someone trying this will want to replace their system-wide `ssh` right away :-)

File: README.md (hunk 2 above)
Comment (xvzcf) [reply README.md]: Added a parenthetical comment.

File: README.md (hunk 5 above)
Comment (xvzcf) [reply README.md]: Right on both counts.

**Comment by xvzcf** [comment]
> Generally, using the regress data works, of course; but I'd then also suggest adding that
> 
>     * host and client keys can be separately generated using `ssh-keygen`

Added a brief note about this.

> 
>     * configuration options (like port etc.) can either be set on the command line or in the respective `ssh_config` and `sshd_config` files in the `regress` folder.
> 

Same here.

> Also, if the whole build documentation is reduced to running the build scripts, I'd suggest adding that "PREFIX" can be set to facilitate a specific `oqs-openssh` install location: I can't imagine someone trying this will want to replace their system-wide `ssh` right away :-)

The `oqs-scripts/build_openssh.sh` script already has an [`INSTALL_PREFIX`](https://github.com/open-quantum-safe/openssh/blob/c518f477f652c903a23cab0ceeca78b977caed60/oqs-scripts/build_openssh.sh#L11) variable, set by default to a non-system location. The README instructions (and the test scripts) however don't use this install location, and simply use the default location of the built artifacts.

File: README.md [hunk 6]
```diff
@@ -112,115 +112,82 @@ On **Ubuntu**, you need to install the following packages:
...
-	                            -o AuthorizedKeysFile=<absolute-path-to>/ssh_server/authorized_keys \
-	                            -o HostKeyAlgorithms=ssh-<SIG> \
-	                            -o PubkeyAcceptedKeyTypes=ssh-<SIG> \
-	                            -h <absolute-path-to>/ssh_server/id_<SIG>
+	<OPENSSH_SRC>/sshd -D \
+	                   -f <OPENSSH_SRC>/regress/sshd_config \
+	                   -o KexAlgorithms=<KEX> \
+	                   -o HostKeyAlgorithms=ssh-<SIG> \
+	                   -o PubkeyAcceptedKeyTypes=ssh-<SIG> \
+	                   -h <absolute-path-to>/regress/host.ssh-<SIG>
```
Comment (baentsch) [inline README.md]: also change `-h` option to OPENSSH_SRC

**Review by baentsch — APPROVED** [review]
Thanks for the updates. LGTM.

**Pull request merged**